plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -494,100 +493,90 @@ public void testMySqlDatetimeType(ZoneId sessionZone)
 
                 // round down
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.123451'", createTimestampType(3), "TIMESTAMP '1970-01-01 00:00:00.123'")
+
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.1234560'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:01.123456'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.1234561'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:00.123456'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.12345649999'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:01.123456'")
 
                 // nanos round up, end result rounds down
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.123499'", createTimestampType(3), "TIMESTAMP '1970-01-01 00:00:00.123'")
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.123399'", createTimestampType(3), "TIMESTAMP '1970-01-01 00:00:00.123'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.123456499'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:00.123456'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.123456499999'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:00.123456'")
 
                 // round up
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.123999'", createTimestampType(3), "TIMESTAMP '1970-01-01 00:00:00.124'")
-
-                // max precision
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.123456'", createTimestampType(3), "TIMESTAMP '1970-01-01 00:00:00.123'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.1234565'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:00.123457'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.1234569'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:01.123457'")
 
                 // round up to next second
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.999999'", createTimestampType(3), "TIMESTAMP '1970-01-01 00:00:01.000'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:00.9999999'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:01.000000'")
 
                 // round up to next day
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 23:59:59.999999'", createTimestampType(3), "TIMESTAMP '1970-01-02 00:00:00.000'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 23:59:59.9999999'", createTimestampType(6), "TIMESTAMP '1970-01-02 00:00:00.000000'")
+
+                // negative epoch - round down
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1969-12-31 23:59:59.1230001'", createTimestampType(6), "TIMESTAMP '1969-12-31 23:59:59.123000'")
```

Comment:
the value being inserted has 7 digits, so it gets rounded/truncated on MySQL side during insert, which is test data preparation step.

i don't think we need to test MySQL INSERT round/truncation behvior. _When inserting into MySQL_ we should use a value that will fit unmodified into the target column type.

Reply:
I will keep only inserted values supported by mysql without any truncation/rounding